### PR TITLE
Fix User Profile icon issue of navbar

### PR DESCRIPTION
**Pull request by Isha988**
## Related Issue
follow up for https://github.com/anmode/grabtern-frontend/issues/322#issuecomment-1638178939

## Description of Changes
1. Added profile Icon and drop down in navbar
2. Fix `be a mentor` button of banner and made it dynamic.

### CODE REVIEW

**Comment by Isha988** [comment]
@anmode  I didn't get the drop down component so created a new one.
Please test these changes at least for one, and if any changes are required so let me know.

File: components/layout/Header.js [hunk 1]
```diff
@@ -159,20 +164,24 @@ function Header() {
...
-                >
-                  Signup
-                </button>
+
+                {/* show profile card if user is logged in else show signup button */}
+                {isUserLoggedIn || isMentorLoggedIn ? (
+                  <UserProfile />
+                ) : (
+                  <ButtonLink text="Sign Up" href="/" />
+                )}
```
Comment (anmode) [inline components/layout/Header.js:184]: For mobile: can we have profile icon , not in drop down... instead near the hamburger. with the same behavior when click on profile that card appears.
Screenshot 2023-07-20 at 12 35 06 PM

**Review by anmode — CHANGES_REQUESTED** [review]
All good, Just a minor change..

**Comment by anmode** [comment]
> @anmode I didn't get the drop down component so created a new one. Please test these changes at least for one, and if any changes are required so let me know.

oh! That's cool you created new one.. it aligned with the theme too. Thanks. Ya, no worries. I'll add the functionality as those are remaining I see. 
like as on logout : `localstorage` should be clear.

File: components/layout/Header.js (hunk 1 above)
Comment (Isha988) [reply components/layout/Header.js:184]: Do we need signup button near hamburger?

Comment (anmode) [reply components/layout/Header.js]: > Do we also need signup button near hamburger?

No no...it's ok in hamburger menu

**Comment by anmode** [comment]
Some last fixes:
1. Nav bar widht should be equal to viewport widht
Screenshot 2023-07-21 at 9 18 46 AM

2. On clicking the humbergur menu, the nav color should changes to white, as it does on scrolling
Screenshot 2023-07-21 at 9 19 30 AM

**Comment by Isha988** [comment]
> Some last fixes:

Done 👍

**Comment by anmode** [comment]
pr ab yeh nav bar scroll ni ho rha :(

and are you available for meet at 9pm today? woh agreement complete krte h.

**Comment by anmode** [comment]
Cool thanks.. and I scheduled the meet :)

**Review by anmode — APPROVED** [review]

**Comment by anmode** [comment]
Now you can carry on...css removal.

**Pull request merged**